internal/pkg/manifest/env.go

```diff
@@ -95,9 +95,47 @@ type environmentNetworkConfig struct {
 }
 
 type environmentVPCConfig struct {
-	ID      *string              `yaml:"id,omitempty"`
-	CIDR    *IPNet               `yaml:"cidr,omitempty"`
-	Subnets subnetsConfiguration `yaml:"subnets,omitempty"`
+	ID                  *string              `yaml:"id,omitempty"`
+	CIDR                *IPNet               `yaml:"cidr,omitempty"`
+	Subnets             subnetsConfiguration `yaml:"subnets,omitempty"`
+	SecurityGroupConfig securityGroupConfig  `yaml:"security_group,omitempty"`
+}
+
+type securityGroupConfig struct {
+	Ingress []securityGroupRules `yaml:"ingress,omitempty"`
+	Egress  []securityGroupRules `yaml:"egress,omitempty"`
+}
+
+// IsEmpty returns true if there is no security group rule defined.
+func (sgc securityGroupConfig) IsEmpty() bool {
+	return len(sgc.Ingress) == 0 && len(sgc.Egress) == 0
+}
+
+type securityGroupRules struct {
+	CidrIp     string `yaml:"cidr"`
+	FromPort   int    `yaml:"from_port"`
+	IpProtocol string `yaml:"ip_protocol"`
+	ToPort     int    `yaml:"to_port"`
+}
+
+//IsValid checks for valid security group config.
+func (sgc securityGroupConfig) IsValid() bool {
```

Comment:
I don't get this method 😅 if the user has set any configuration we return false?

Reply:
Hmm, actually I was using condition in [here](https://github.com/aws/copilot-cli/pull/3810/files#diff-b2b0270288b7891d9a0645336e6267dd39f5b3f4a6b3bf1f6ef19bd4c2806195R203) which was making sense because of negation but the usage was not right.  I have changed it to read it correctly now?